src/realm/object-store/property.hpp

```diff
@@ -238,7 +238,7 @@ static auto switch_on_type(PropertyType type, Fn&& fn)
         case PT::Date:
             return fn((Timestamp*)0);
         case PT::Object:
-            return fn((ObjType*)0);
+            return is_optional ? fn((Mixed *) 0) : fn((ObjType *) 0);
```

Comment:
I don't think this is in line with the general approach in this function. The idea is that `fn` should be called with a parameter of the type given by `type`. This change breaks this contract. It might be that the change fixes the issue here, but this is because Dictionary uses Mixed as the only type in the interface.

Reply:
Yes, it did not make sense for the other data types.